xbmc/cores/VideoRenderers/BaseRenderer.cpp

```diff
@@ -222,11 +222,20 @@ void CBaseRenderer::FindResolutionFromFpsMatch(float fps, float& weight)
 RESOLUTION CBaseRenderer::FindClosestResolution(float fps, float multiplier, RESOLUTION current, float& weight)
 {
   RESOLUTION_INFO curr = g_graphicsContext.GetResInfo(current);
+  RESOLUTION orig_res  = CDisplaySettings::GetInstance().GetCurrentResolution();
+
+  if (orig_res <= RES_DESKTOP)
+    orig_res = RES_DESKTOP;
+
+  RESOLUTION_INFO orig = g_graphicsContext.GetResInfo(orig_res);
 
   float fRefreshRate = fps;
 
   float last_diff = fRefreshRate;
 
+  int curr_diff = std::abs((int) m_sourceWidth - curr.iScreenWidth);
```

Comment:
don't you miss some parenthesis? you cast the first parameter to int, then subtract float.